build_tools/scripts/run_markdownlint.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Runs Markdownlint on Markdown (.md) files
+# https://github.com/igorshubovych/markdownlint-cli
+
+set -xuo pipefail
+
+declare -a included_files_patterns=(
+  # All .md files (disabled while we decide how rigorously to apply lint checks)
+  # "./**/*.md"
+
+  # Just .md files for the user-facing website.
+  "./docs/website/**/*.md"
+)
+
+declare -a excluded_files_patterns=(
+  "/third_party/"
+  "**/node_modules/**"
+)
```

Comment:
Note: if you have local markdown files and want to explicitly exclude them too (I have untracked `iree/notes/*.md` files, for example), you can add a `.markdownlintignore` file to any directory with contents like
```gitignore
./**/*.md
```

Reply:
Should we use `.markdownlintignore` instead of these exclusions then?

Reply:
* I didn't want to add a top level `.markdownlintignore` next to our `.gitignore`
* I want to exclude `third_party` and `node_modules` across the entire project (some of my work on webgpu/wasm/emscripten created some node_modules folders on my branches, we don't have any merged to `main` yet... but we definitely don't want to be linting node_modules :P)
* Our other lint scripts use exclude patterns in this style, so matching them is nice
* The ignore files and patterns can be mixed